frame/multi-account/src/lib.rs

```diff
@@ -0,0 +1,1506 @@
+			ensure!(other_signatories.len() < max_sigs, Error::<T>::TooManySignatories);
+			ensure!(other_signatories.len() < u32::max_value() as usize, Error::<T>::TooManySignatories);
+			let signatories = Self::ensure_sorted_and_insert(other_signatories.clone(), who.clone())?;
+
+			let multi_account_index = MultiAccountIndex::get()
+				.checked_add(1)
+				.expect("multi account indices will never reach 2^64 before the death of the universe; qed");
+
+			let id = Self::multi_account_id(multi_account_index);
+
+			let deposit = T::MultiAccountDepositBase::get()
+				+ T::MultiAccountDepositFactor::get() * (other_signatories.len() as u32 + 1).into();
+
+			MultiAccountIndex::put(multi_account_index);
+
+			T::Currency::reserve(&who, deposit)?;
+
+			<MultiAccounts<T>>::insert(&id, MultiAccountData {
+				threshold,
+				signatories,
+				deposit,
+				depositor: who.clone(),
+			});
+
+			Self::deposit_event(RawEvent::NewMultiAccount(who, id));
+
+			Ok(())
+		}
+
+		/// Update an existing multi account with a new threshold and new signatories.
+		///
+		/// This maintains the multi account's account id.
+		///
+		/// Payment: `MultiAccountDepositBase` will be reserved, plus the number of signatories times
+		/// `MultiAccountDepositFactor`. It is returned when the multi account is updated or removed. This call will
+		/// unreserve the previous deposit.
+		///
+		/// The dispatch origin for this call must be _Signed_ and be equal to the multi account ID. It can be
+		/// dispatched by a `call` from this module.
+		///
+		/// Any removed signatories that have approved multisig transactions will no longer be counted as valid
+		/// approvals. Removing a signatory that has initiated/is the owner of a multisig transaction will not
+		/// cancel the multisig transaction, but the owner can always cancel the multisig transaction, even after
+		/// leaving the signatories. Such a transaction can still be dispatched, as long as the threshold
+		/// is met by the current signatories.
+		///
+		/// - `threshold`: The total number of approvals dispatches from this multi account require before they are
+		///    executed.
+		/// - `signatories`: The accounts who can approve dispatches from this multi account. May not be empty.
+		///
+		/// # <weight>
+		/// - `O(S)`.
+		/// - One balance-reserve and unreserve operation.
+		/// - One insert with `O(S)` where `S` is the number of
+		///   signatories. `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - I/O: 1 read `O(S)`, 1 mutate `O(S)`.
+		/// - One event.
+		/// - Storage: inserts one item, value size bounded by `MaxSignatories`, with a
+		///   deposit taken for its lifetime of
+		///   `MultiAccountDepositBase + signatories.len() * MultiAccountDepositFactor`.
+		/// # </weight>
+		#[weight = <SigsLen<T::AccountId>>::new()]
+		fn update(origin,
```

Comment:
I made a comment below, but the assumptions in this pallet can be reduced if you ensure that there are no existing multisig operations when update is called.

You can choose to have this function fail or you can chose to automatically delete any existing operations.

Reply:
You can't really delete any existing operations, so you should fail here.

Reply:
See discussion above